### PR TITLE
Adding deployment script for frontend application

### PR DESCRIPTION
**Pull request by tushardhadiwal**
This PR adds following :
- A fully automated deployment script that creates resources in azure, builds and deploys frontend application to Azure web app, sets the correct environment variables as app settings on web app.
- This also adds azure Active Directory authentication to the frontend web app using federated identity credentials.
- Post deployment the redirect URL of the deployed web app is added to app registration. 
- It also includes a function that enables outbound vent integration on the web app resource to enable communication graph rag backend deployed with peered vnets. (After the simplify networking PR is merged, this will not be default behavior, so that function is commented.)
- ReadMe.md file has been updated to reflect steps for creating app registration and pre deployment configuration details.

### CODE REVIEW

File: frontend/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+}
+
+function create_acr() {
+    az acr create --resource-group $RESOURCE_GROUP \
+    --name $REGISTRY_NAME \
+    --sku Basic \
+    --admin-enabled true
+}
+
+function login_to_acr() {
```
Comment (jgbradley1) [inline frontend/deploy.sh:66]: I don't believe this step is needed as I was able to comment out the function and script still built and pushed the docker image into the container registry. This step also seemed to be the only place in the script that required docker to be installed.

File: frontend/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    WEBAPP_IDENTITY_OBJECT_ID=$(jq -r .principalId <<< $IDENTITY_RESULT)
+    WEBAPP_IDENTITY_CLIENT_ID=$(jq -r .clientId <<< $IDENTITY_RESULT)
+    az webapp identity assign --name $WEB_APP \
+        --resource-group $RESOURCE_GROUP \
+        --identities $WEBAPP_IDENTITY_ID
+}
+
+function configure_registry_credentials() {
+    ACR_ID=$(az acr show --name $REGISTRY_NAME --resource-group $RESOURCE_GROUP --query id --output tsv)
+    az role assignment create --assignee $WEBAPP_IDENTITY_CLIENT_ID \
```
Comment (jgbradley1) [inline frontend/deploy.sh:101]: I get the following error at this step in the script

```
Cannot find user or service principal in graph database for '<webapp_identity_client_idenity>`. If the assignee is an appId, make sure the corresponding service principal is created with 'az ad sp create --id '<webapp_identity_client_idenity>'.'
```
Except the webapp client identity is filled out and correct. I verified in the portal.

File: frontend/deploy.sh (hunk 1 above)
Comment (tushardhadiwal) [reply frontend/deploy.sh:66]: We can remove the `login_to_acr` function if it's not needed.

File: frontend/deploy.sh (hunk 2 above)
Comment (tushardhadiwal) [reply frontend/deploy.sh:112]: Issue identified as `az-cli` configuration not outputting json as needed for WEBAPP_IDENTITY_CLIENT_ID. Script updated to force json output. Issue resolved.

**Review by jgbradley1 — APPROVED** [review]

**Pull request merged**